build/ci: update golangci-linter with --new-from-rev

Muzry opened the pull request:
By using `--new-from-rev=origin/master` with golangci-lint, it only lint the files that have changed compared to `origin/master`, reducing the linting workload.

MariusVanDerWijden commented:
level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing \"git diff --color=never --no-ext-diff --default-prefix --relative origin/master --\": exit status 128: fatal: bad revision 'origin/master'\n"

As I expected, this does not work in our CI

Muzry commented:
> level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing "git diff --color=never --no-ext-diff --default-prefix --relative origin/master --": exit status 128: fatal: bad revision 'origin/master'\n"
> 
> As I expected, this does not work in our CI

@MariusVanDerWijden Thanks for your reply and for helping fix the branch issue. I have updated the configs and fixed the problem. Please review again.

MariusVanDerWijden commented:
Its still not faster... The lint was 5 min before and is still 5 min now. I don't think it makes sense to add this, sorry. We always want the linter to run on all the files in the repository, since renamings can affect non-changed files and this would miss that
